### PR TITLE
RUMM-1472 nightly test coverage

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Update nightly test coverage computation: 
- Use canonical name to avoid ambiguous class name being ignored
- Fix nested class name computation
- Ignore data classes, enums and interfaces by default

### CODE REVIEW

File: instrumentation-tools/src/api_surface.py [hunk 1]
```diff
@@ -2,15 +2,18 @@
...
-DIVIDER = "#"
 DOT = "."
+NESTED_TYPE_SEPARATOR = "$"
+API_SEPARATOR = "#"
 EXTEND_SEPARATOR = ":"
-TESTABLE_API_REGEX = re.compile("^(?<!deprecated)(\\s*)(constructor|fun)(.+)", re.IGNORECASE)
-TYPE_REGEX = re.compile("(\\s*)(.*)(class|interface|enum|object)(\\s)(.+)$", re.IGNORECASE)
-DEFAULT_IGNORED_TYPE_REGEX = re.compile("(.*)(companion|deprecated)(\\s)(.*)$", re.IGNORECASE)
+TESTABLE_API_REGEX = re.compile("^(\\s*)(?<!deprecated)(\\s*)(constructor|fun)(.+)")
+TYPE_REGEX = re.compile("(\\s*)(.*)(class|interface|enum|object)(\\s)(.*)$")
```
Comment (0xnm) [inline instrumentation-tools/src/api_surface.py:14]:
```suggestion
TYPE_REGEX = re.compile("^(\\s*)(.*)(class|interface|enum|object)(\\s)(.+)$")
```

**Review by 0xnm — APPROVED** [review]

File: instrumentation-tools/src/api_surface.py [hunk 2]
```diff
@@ -61,43 +44,46 @@ def sanitize_prefix(previous_indent: int,
...
         # We will need this prefix which holds also the attributes of the type (e.g. "companion", "deprecated", "open")
         # to be able to ignore some APIs.
-        prefix_with_type_attributes = ""
+        type_attributes = ""
+        type = ""
 
-        current_indent = 0
         for row in rows:
             is_type_matcher = TYPE_REGEX.match(row)
+            is_testable_api = TESTABLE_API_REGEX.match(row)
```
Comment (mariusc83) [inline instrumentation-tools/src/api_surface.py:70]: This code is quite complex, maybe a good idea would be (something that I should have added when I first wrote this) is to add some unit tests. Maybe we can open a new task for this and we can add those next sprint (will be glad to handle that).

**Review by mariusc83 — APPROVED** [review]
Nice work !

File: instrumentation-tools/src/api_surface.py [hunk 3]
```diff
@@ -61,43 +44,46 @@ def sanitize_prefix(previous_indent: int,
...
-            elif TESTABLE_API_REGEX.match(row) \
-                    and (not DEFAULT_IGNORED_TYPE_REGEX.match(prefix_with_type_attributes)) \
-                    and (self.extra_ignored_types_reg_ex is None or not self.extra_ignored_types_reg_ex.match(prefix)):
-                to_return.add(prefix + row.strip())
+                stack[depth] = type_name
+            elif is_testable_api:
+                depth = int(len(is_testable_api.group(1)) / INDENT_SIZE)
+                prefix = NESTED_TYPE_SEPARATOR.join(stack[:depth])
+                if (type_attributes in DEFAULT_IGNORED_TYPE_ATTRS) or (type in DEFAULT_IGNORED_TYPES):
+                    # print(f"Ignored by default {type_attributes} {type} {prefix}{API_SEPARATOR}{row.strip()}")
```
Comment (mariusc83) [inline instrumentation-tools/src/api_surface.py:82]: ?

File: instrumentation-tools/src/api_surface.py [hunk 4]
```diff
@@ -61,43 +44,46 @@ def sanitize_prefix(previous_indent: int,
...
+                    pass
+                elif self.ignored_types is not None and prefix in self.ignored_types:
+                    print(f"Ignored manually {type_attributes} {type} {prefix}{API_SEPARATOR}{row.strip()}")
+                else:
+                    # print(f"Keeping {type_attributes} {type} {prefix}{API_SEPARATOR}{row.strip()}")
```
Comment (mariusc83) [inline instrumentation-tools/src/api_surface.py:87]: ?

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: instrumentation-tools/src/api_surface.py (hunk 1 above)
Comment (xgouchet) [reply instrumentation-tools/src/api_surface.py:14]: This `*` is on purpose in order to catch `companion object` declarations.

```
// note there's a trailing space in the following line
      companion object 
(\s*)|  (.*)   |object|(\s)|(.*)
```

File: instrumentation-tools/src/api_surface.py (hunk 2 above)
Comment (xgouchet) [reply instrumentation-tools/src/api_surface.py:70]: What's complex about it ?

Comment (mariusc83) [reply instrumentation-tools/src/api_surface.py:70]: well it much more easier to update it later with unit tests in place.

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**